src/test/java/com/synopsys/integration/detect/battery/docker/integration/RapidModeTests.java

```diff
@@ -18,23 +18,25 @@
 @Tag("integration")
 public class RapidModeTests {
     SharedDockerTestRunner anyProjectWithRapidResultsInBlackDuck(String testId, NameVersion projectNameVersion) throws IOException, IntegrationException {
-        DetectDockerTestRunner runner = new DetectDockerTestRunner(testId, "gradle-simple:1.0.0");
-        runner.withImageProvider(BuildDockerImageProvider.forDockerfilResourceNamed("SimpleGradle.dockerfile"));
+        // DetectDockerTestRunner must be declared in try-with-resources block to take advantage of "close" method (cleans up resources)
+        try (DetectDockerTestRunner runner = new DetectDockerTestRunner(testId, "gradle-simple:1.0.0")) {
+            runner.withImageProvider(BuildDockerImageProvider.forDockerfilResourceNamed("SimpleGradle.dockerfile"));
 
-        BlackDuckTestConnection blackDuckTestConnection = BlackDuckTestConnection.fromEnvironment();
-        BlackDuckAssertions blackduckAssertions = blackDuckTestConnection.projectVersionAssertions(projectNameVersion);
-        blackduckAssertions.emptyOnBlackDuck();
+            BlackDuckTestConnection blackDuckTestConnection = BlackDuckTestConnection.fromEnvironment();
+            BlackDuckAssertions blackduckAssertions = blackDuckTestConnection.projectVersionAssertions(projectNameVersion);
+            blackduckAssertions.emptyOnBlackDuck();
 
-        DetectCommandBuilder commandBuilder = new DetectCommandBuilder().defaults().defaultDirectories(runner);
-        commandBuilder.connectToBlackDuck(blackDuckTestConnection);
-        commandBuilder.projectNameVersion(blackduckAssertions);
-        commandBuilder.tools(DetectTool.DETECTOR); //All that is needed for a BOM in black duck.
+            DetectCommandBuilder commandBuilder = new DetectCommandBuilder().defaults().defaultDirectories(runner);
+            commandBuilder.connectToBlackDuck(blackDuckTestConnection);
+            commandBuilder.projectNameVersion(blackduckAssertions);
+            commandBuilder.tools(DetectTool.DETECTOR); //All that is needed for a BOM in black duck.
 
-        return new SharedDockerTestRunner(runner, blackDuckTestConnection, blackduckAssertions, commandBuilder);
+            return new SharedDockerTestRunner(runner, blackDuckTestConnection, blackduckAssertions, commandBuilder);
```

Comment:
No issues with returning the try-with-resources here? Is it closed after it's garbage collected? I don't quite understand how this worked.

Reply:
Hmm that's a good question...I guess I don't know either.